usecases/replica/coordinator.go

```diff
@@ -190,38 +213,86 @@ func (c *coordinator[T]) Pull(ctx context.Context,
 	}
 	level := state.Level
 	replyCh := make(chan _Result[T], level)
-
-	candidates := state.Hosts[:level]                          // direct ones
-	candidatePool := make(chan string, len(state.Hosts)-level) // remaining ones
-	for _, replica := range state.Hosts[level:] {
-		candidatePool <- replica
-	}
-	close(candidatePool) // pool is ready
+	hosts := state.Hosts
 	f := func() {
+		hostRetryQueue := make(chan hostRetry, len(hosts))
+
+		// put the "backups/fallbacks" on the retry queue
+		for i := level; i < len(hosts); i++ {
```

Comment:
Really like how you changed the retry logic here, great improvement over our first pair session on it. Good job :)